main/guides/js-programming/far.md

```diff
@@ -28,13 +28,39 @@ In particular, note that every object returned from a smart contract, such a `pu
 `creatorFacet`, must be passable. All objects used in your contract's external API must
 be passable.
 
+### Rules for creating passable objects
+Record and tuples are hardened acyclical objects which can be passed-by-copy if they satisfy some constraints.
+
+#### Record
+A plain object where:
+- All property values must be passables.
+  - Properties cannot be accessors.
+- All properties must be own and enumerable, with string names.
+  - Symbols are not allowed.
+  - The prototype must be `Object.prototype` or `null`.
+- Nested records and tuples cannot form a cycle.
+- It must be hardened with `harden()`.
+
+#### Tuple
+A plain array where:
+- All property values must be passables.
+  - Properties cannot be accessors.
+- All properties must be own, and be all the integers indexes or `length`.
```

Comment:
```suggestion
- All properties must be own, and contain all the integers indexes up to `length`.
```

Reply:
Technically `length` is also an own property of the array, and the array cannot have non indexes properties (besides `length`). I know my formulation is a bit clunky, but I believe it is more correct, as "contain" doesn't preclude extra props. I would love to find better wording, but I think we should strive to keep it correct.

Reply:
I couldn't parse "and be all the integers indexes or `length`."   Is the requirement that all integers up to `length - 1` must be present,  or only that nothing else is allowed, or something else?

Reply:
I like @Chris-Hibbert 's suggestion. I'd add "No other properties are allowed" (unless that is already implied by the surrounding text).

Reply:
"present" how? As indices? If so, specify that.